packages/eslint-plugin/src/rules/no-unsafe-assignment.ts

```diff
@@ -251,8 +251,8 @@ export default createRule({
       const receiverTsNode = services.esTreeNodeToTSNodeMap.get(receiverNode);
       const receiverType =
         comparisonType === ComparisonType.Contextual
-          ? getContextualType(checker, receiverTsNode as ts.Expression) ??
-            services.getTypeAtLocation(receiverNode)
+          ? (getContextualType(checker, receiverTsNode as ts.Expression) ??
+            services.getTypeAtLocation(receiverNode))
```

Comment:
https://github.com/prettier/prettier/blob/HEAD/CHANGELOG.md#add-parentheses-for-nullish-coalescing-in-ternary-16391-by-cdignam-segment again